src/posthog-surveys.ts

```diff
@@ -214,34 +221,50 @@ export class PostHogSurveys {
             return
         }
 
-        const generateSurveys = phExtensions.generateSurveys
-
         if (!this._decideServerResponse) {
             logger.warn('Decide not loaded yet. Not loading surveys.')
             return
         }
 
-        if (this._surveyEventReceiver == null) {
-            this._surveyEventReceiver = new SurveyEventReceiver(this.instance)
-        }
-
-        if (!generateSurveys) {
-            const loadExternalDependency = phExtensions.loadExternalDependency
-
-            if (loadExternalDependency) {
-                loadExternalDependency(this.instance, 'surveys', (err) => {
-                    if (err) {
-                        logger.error('Could not load surveys script', err)
-                        return
-                    }
-
-                    this._surveyManager = phExtensions.generateSurveys?.(this.instance)
-                })
+        this._isInitializingSurveys = true
+
+        try {
+            const generateSurveys = phExtensions.generateSurveys
+
+            if (!generateSurveys) {
+                const loadExternalDependency = phExtensions.loadExternalDependency
+
+                if (loadExternalDependency) {
+                    loadExternalDependency(this.instance, 'surveys', (err) => {
+                        if (err) {
+                            logger.error('Could not load surveys script', err)
+                            this._isInitializingSurveys = false
+                            return
+                        }
+
+                        this._surveyManager = phExtensions.generateSurveys?.(this.instance)
+                        this._isInitializingSurveys = false
+                        if (this._surveyEventReceiver == null) {
+                            this._surveyEventReceiver = new SurveyEventReceiver(this.instance)
+                        }
```

Comment:
do we actually need this guard? the whole control flow now is protected, will this method ever be called again if `_surveyManager` was created and consequently `_surveyEventReceiver`?

Reply:
good point, no need for it anymore. removed it